comp/core/tagger/types/entity_id.go

```diff
@@ -0,0 +1,148 @@
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package types defines types used by the Tagger component.
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+// EntityID represents a tagger entityID
+// An EntityID should be identified by a prefix and an id, and is represented as {prefix}://{id}
+type EntityID interface {
+	// GetID returns a prefix-specific id (i.e. an ID unique given prefix)
+	GetID() string
+	// GetPrefix returns the prefix of the EntityID
+	GetPrefix() EntityIDPrefix
+	// String returns a string representation of EntityID under the format {prefix}://{id}
+	String() string
+}
+
+// defaultEntityID implements EntityID as a plain string id
+type defaultEntityID string
+
+// GetID implements EntityID#GetID
+func (de defaultEntityID) GetID() string {
+	if strings.Contains(string(de), "://") {
+		return string(de)
+	}
+
+	parts := strings.Split(string(de), "://")
+	if len(parts) != 2 {
+		return ""
+	}
+
+	return parts[0]
+}
+
+// GetPrefix implements EntityID#GetPrefix
+func (de defaultEntityID) GetPrefix() EntityIDPrefix {
+	if strings.Contains(string(de), "://") {
+		return EntityIDPrefix(de)
+	}
+
+	parts := strings.Split(string(de), "://")
+	if len(parts) != 2 {
+		return ""
+	}
+
+	return EntityIDPrefix(parts[0])
+}
+
+// String implements EntityID#String
+func (de defaultEntityID) String() string {
+	return string(de)
+}
+
+func newDefaultEntityID(id string) EntityID {
+	return defaultEntityID(id)
+}
+
+// compositeEntityID implements EntityID as a struct of prefix and id
+type compositeEntityID struct {
+	Prefix EntityIDPrefix
+	ID     string
+}
+
+// GetPrefix implements EntityID#GetPrefix
+func (eid compositeEntityID) GetPrefix() EntityIDPrefix {
+	return eid.Prefix
+}
+
+// GetID implements EntityID#GetID
+func (eid compositeEntityID) GetID() string {
+	return eid.ID
+}
+
+// String implements EntityID#String
+func (eid compositeEntityID) String() string {
+	return eid.Prefix.ToUID(eid.ID)
+}
+
+// newcompositeEntityID returns a new EntityID based on a prefix and an id
+func newCompositeEntityID(prefix EntityIDPrefix, id string) EntityID {
+	return compositeEntityID{
+		Prefix: prefix,
+		ID:     id,
+	}
+}
+
+// NewEntityID builds and returns an EntityID object based on plain string uid
+// Currently, it defaults to the default implementation of EntityID as a plain string
+func NewEntityID(prefix EntityIDPrefix, id string) EntityID {
+	if config.Datadog().GetBool("tagger.tagstore_use_composite_entity_id") {
+		return newCompositeEntityID(prefix, id)
+	}
+	return newDefaultEntityID(fmt.Sprintf("%s://%s", prefix, id))
+}
+
+// NewEntityIDFromString constructs EntityID from a plain string id
+func NewEntityIDFromString(plainStringID string) (EntityID, error) {
+	if config.Datadog().GetBool("tagger.tagstore_use_composite_entity_id") {
```

Comment:
Could you use the config component rather than accessing the config directly using `pkg/config`?

Reply:
Ideally, we don't need to access the config at all here.

This is temporary just to be able to switch between the 2 implementations while we implement filters and finish things up.

Once it is done, the composite store should be used always. 

I didn't use the component framework here just to make it (slightly easier), to call these methods and setup the store. 

If you think it is still better to use the component framework here, I can still do the change.

WDYT?

Reply:
> If you think it is still better to use the component framework here, I can still do the change.

I think it does. For the main reason that if the temporary solution endup taking longer to implement of for whatever reason we stay in the state of supporting the two solutions, at least we are not introducing the anti pattern of using `pkg/config` inside `comp` folder. 

And on the bright side if you pass the config component once you are ready to use the composite store everywhere you would not have to add the config component to the composite store 😄

Reply:
Discussed on slack, and agreed on keeping it the way it is to avoid refactoring code of many other teams calling the tagger.

Added a TODO comment to not forget to remove it, and created a [JIRA card.](https://datadoghq.atlassian.net/jira/software/c/projects/CONTP/boards/6366?assignee=712020%3A21ad1fcc-72ec-42e0-8e00-b062b596303e&selectedIssue=CONTP-388)